tests/test_cli.py

```diff
@@ -0,0 +1,286 @@
+        assert main_under_test(["example.com", "status"]) == 1
+
+    assert out.getvalue() == ""
+    assert err.getvalue() == "Error: \n"
+
+
+def test_query(mock_network_requests):
+    with patch_stdout_stderr() as (out, err):
+        assert main_under_test(["example.com", "query"]) == 0
+
+    assert (
+        "host: 192.168.56.1:9999\n"
+        "software: Java 1.8 vanilla\n"
+        "motd: \x1b[0mA Minecraft Server\x1b[0m\n"
+        "plugins: []\n"
+        "players: 3/20 ['Dinnerbone', 'Djinnibone', 'Steve']\n"
+    ) == out.getvalue()
+    assert err.getvalue() == ""
+
+
+def test_query_offline(mock_network_requests):
+    with patch_stdout_stderr() as (out, err), patch("mcstatus.server.JavaServer.query", side_effect=socket.timeout):
+        assert main_under_test(["example.com", "query"]) != 0
+
+    assert out.getvalue() == ""
+    assert err.getvalue() == QUERY_FAIL_WARNING + "\n"
+
+
+def test_json(mock_network_requests):
+    with patch_stdout_stderr() as (out, err):
+        assert main_under_test(["example.com", "json"]) == 0
+
+    data = json.loads(out.getvalue())
+    assert data == {
+        "online": True,
+        "kind": "Java",
+        "status": {
+            "players": {"online": 0, "max": 20, "sample": None},
+            "version": {"name": "1.8-pre1", "protocol": 44},
+            "motd": "A Minecraft Server",
+            "latency": 0,
+            "raw": {
+                "players": {"max": 20, "online": 0},
+                "version": {"name": "1.8-pre1", "protocol": 44},
+                "description": "A Minecraft Server",
+                "enforcesSecureChat": True,
+                "favicon": "data:image/png;base64,foo",
+            },
+            "enforces_secure_chat": True,
+            "icon": "data:image/png;base64,foo",
+            "forge_data": None,
+        },
+        "query": {
+            "ip": "192.168.56.1",
+            "port": "9999",
+            "map": "world",
+            "plugins": [],
+            "raw": {
+                "hostname": "A Minecraft Server",
+                "gametype": "GAME TYPE",
+                "game_id": "GAME ID",
+                "version": "1.8",
+                "plugins": "",
+                "map": "world",
+                "numplayers": "3",
+                "maxplayers": "20",
+                "hostport": "9999",
+                "hostip": "192.168.56.1",
+            },
+        },
+    }
+    assert err.getvalue() == ""
+
+
+def test_ping(mock_network_requests):
+    with patch_stdout_stderr() as (out, err):
+        assert main_under_test(["example.com", "ping"]) == 0
+
+    assert float(out.getvalue()) == 0
+    assert err.getvalue() == ""
+
+
+def test_ping_bedrock(mock_network_requests):
+    with patch_stdout_stderr() as (out, err):
+        assert main_under_test(["example.com", "--bedrock", "ping"]) == 0
+
+    assert float(out.getvalue()) == 0
+    assert err.getvalue() == ""
```

Comment:
This doesn't guarantee that the bedrock logic was ran here, this test would pass if the CLI just ignored the `--bedrock` argument entirely. Either check that certain patched function was called, or make the `mock_network_requests` return a different latency for bedrock

Reply:
Done, thanks!